src/MiddlewareDispatcher.php

```diff
@@ -87,13 +87,13 @@ public function hasMiddlewares(): bool
     }
 
     /**
-     * @return MiddlewareInterface[]
+     * @return \Closure[]
      */
     private function buildMiddlewares(): array
     {
         $middlewares = [];
         foreach ($this->middlewareDefinitions as $middlewareDefinition) {
-            $middlewares[] = $this->middlewareFactory->create($middlewareDefinition);
+            $middlewares[] = fn (): MiddlewareInterface => $this->middlewareFactory->create($middlewareDefinition);
```

Comment:
What if `$middlewareDefinition` will be closure like `fn () => new Middleware();`?

Reply:
Any idea on how to do it another way?